DigitalPuzzle/DigitalPuzzle.ino

```diff
@@ -355,61 +244,284 @@ void ReadSerial()
+void RandomizeStartValues()
+{
+	for (int i; i < 6; i++)
+	{
+		StartValues[i] = random() % 2;
+	}
+	wires[0].Type  = (WireType)StartValues[0];
+	wires[1].Type  = (WireType)StartValues[1];
+	wires[3].Type  = (WireType)StartValues[2];
+	wires[4].Type  = (WireType)StartValues[3];
+	wires[11].Type = (WireType)StartValues[4];
+	wires[6].Type  = (WireType)StartValues[5];
+}
+
+//waar is deze functie precies voor?
+void RandomSSNumber()
+{
+  int RandomNumber = 0;
+  RandomNumber = random(0,9);
+
+  switch(RandomNumber)
+  {
+    case 0:
+      digitalWrite(PinA, LOW);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, LOW);
+      digitalWrite(PinD, LOW);
+      break;
+    case 1:
+      digitalWrite(PinA, HIGH);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, LOW);
+      digitalWrite(PinD, LOW);
+      break; 
+    case 2:
+      digitalWrite(PinA, LOW);
+      digitalWrite(PinB, HIGH);
+      digitalWrite(PinC, LOW);
+      digitalWrite(PinD, LOW);
+      break;
+    case 3:
+      digitalWrite(PinA, HIGH);
+      digitalWrite(PinB, HIGH);
+      digitalWrite(PinC, LOW);
+      digitalWrite(PinD, LOW);
+      break; 
+    case 4:
+      digitalWrite(PinA, LOW);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, HIGH);
+      digitalWrite(PinD, LOW);
+      break;
+    case 5:
+      digitalWrite(PinA, HIGH);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, HIGH);
+      digitalWrite(PinD, LOW);
+      break; 
+    case 6:
+      digitalWrite(PinA, LOW);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, HIGH);
+      digitalWrite(PinD, LOW);
+      break;
+    case 7:
+      digitalWrite(PinA, HIGH);
+      digitalWrite(PinB, HIGH);
+      digitalWrite(PinC, HIGH);
+      digitalWrite(PinD, LOW);
+      break; 
+    case 8:
+      digitalWrite(PinA, LOW);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, LOW);
+      digitalWrite(PinD, HIGH);
+      break;
+    case 9:
+      digitalWrite(PinA, HIGH);
+      digitalWrite(PinB, LOW);
+      digitalWrite(PinC, LOW);
+      digitalWrite(PinD, HIGH);
+      break; 
   }
 }
 
 
+void calculateReader(int Reader){
+	switch (CompareNFC(ReadRFID(Reader)))
+	{
+		case 0:
+			readers[Reader] = portReader(portReader::NO_GATE);
+		break;
+		case 1:
+			readers[Reader] = portReader(portReader::AND_GATE);
+		break;
+		case 2:
+			readers[Reader] = portReader(portReader::AND_GATE);
+		break;
+		case 3:
+			readers[Reader] = portReader(portReader::OR_GATE);
+		break;
+		case 4:
+			readers[Reader] = portReader(portReader::NAND_GATE);
+		break;
+		case 5:
+			readers[Reader] = portReader(portReader::NOR_GATE);
+		break;
+		case 6:
+			readers[Reader] = portReader(portReader::XOR_GATE);
+		break;
+		case 7:
+			readers[Reader] = portReader(portReader::INVERTER);
+		break;
+		case 8:
+			readers[Reader] = portReader(portReader::WIRE);
+		break;
+	}
+}
+
+void setup()
+{
+	Serial.begin(115200);
+	randomSeed(analogRead(1));
+	delay(10);
```

Comment:
Waar is dit voor nodig?

Reply:
De delay kan waarschijnlijk ook in de prullenbak ja. Tbh echt 0 idee waar dat aan ligt. Als je de random seed bedoelt.... Die is voor het willekeurig generen voor de startwaardes van de puzzel